libraries/api-cat-client/src/main/java/no/dcat/client/apicat/ApiCatClient.java

```diff
@@ -0,0 +1,49 @@
+package no.dcat.client.apicat;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.client.RestTemplate;
+
+public class ApiCatClient {
+    private String apiCatUrl;
+
+    public ApiCatClient(String apiCatUrl) {
+        this.apiCatUrl = apiCatUrl;
+    }
+
+    public OpenAPI convert(String url, String spec) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        ConvertRequest request = ConvertRequest.builder()
+                .url(url)
+                .spec(spec)
+                .build();
+
+        HttpEntity<ConvertRequest> entity = new HttpEntity<>(request);
+
+        ConvertResponse convertResponse = restTemplate.exchange(this.apiCatUrl + "/convert", HttpMethod.POST, entity, ConvertResponse.class).getBody();
+
+        OpenAPI openAPI = convertResponse.openApi;
+
+        if (openAPI != null) {
+            return openAPI;
+        }
+
```

Comment:
Personlig ville jeg gjort dette motsatt. Altså if (openAPI==null) kast exception først, og så return openAPI til slutt. Litt smak og behag, men det er god vane å unngå for mye return midt inne i metoder.

Reply:
jeg er absolutt enig! kast først!